packages/datagrid/src/datagrid.ts

```diff
@@ -2096,6 +2096,77 @@ export class DataGrid extends Widget {
     MessageLoop.postMessage(this._viewport, Private.OverlayPaintRequest);
   }
 
+  private _getMaxWidthInBodyColumn(index: number): number | null {
+    const dataModel = this.dataModel;
+
+    if (!dataModel) {
+      return null;
+    }
+
+    const numRows = dataModel.rowCount('body');
+    const configs = Array.from({ length: numRows }, (_val, idx) =>
+      DataGrid._getConfig(dataModel, idx, index, 'body')
+    );
+
+    // Heuristic: Sort by the length of the text to render and only fully calculate the text width
```

Comment:
I can decrease the number of selected rows or try using a heap/quickselect for getting the top-N elements instead if this is not adequate.

Reply:
Update: MinHeap won't really help with performance. The main issue seems to be that getting the values of the cells is extremely slow. This is possibly accounted by the fact that the data is stored in a row-major form:

https://github.com/jupyterlab/lumino/blob/9c4933fb2db746346882550fd230238c420f2501/packages/datagrid/src/jsonmodel.ts#L83

I think that what I'm doing here is probably good-enough for now but can change it if you think that this is likely not adequate.